### PR TITLE
[tanium] Fix handling of differently formatted data

### PR DESCRIPTION
**Pull request by chrisberkhout**
## Proposed commit message

```
[tanium] Fix handling of differently formatted data (#)

For the `threat_response` data stream:
- Handle `state` when it's parsed JSON (as well as when it's stringified
  JSON).
- Set `user.id` and `user.related` after processing 'User Id', so its
  value is used.
- Handle `Match Details` data in its own field (the same as when it's in
  an encoded payload).

For all data streams:
- Add processor tags and improve `on_failure` handling.
```

## Discussion

This can be reviewed commit-by-commit.

## Checklist

- [ ] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/main/docs/tips_for_building_integrations.md) and this pull request is aligned with them.
- [ ] I have verified that all data streams collect metrics or logs.
- [x] I have added an entry to my package's `changelog.yml` file.
- [ ] I have verified that Kibana version constraints are current according to [guidelines](https://github.com/elastic/elastic-package/blob/master/docs/howto/stack_version_support.md#when-to-update-the-condition).
- [ ] I have verified that any added dashboard complies with Kibana's [Dashboard good practices](https://docs.elastic.dev/ux-guidelines/data-viz/dashboard-good-practices)

### CODE REVIEW

File: packages/tanium/changelog.yml [hunk 1]
```diff
@@ -1,4 +1,9 @@
 # newer versions go on top
+- version: "1.10.2"
+  changes:
+    - description: Fix handling of differently formatted data
```
Comment (efd6) [inline packages/tanium/changelog.yml:4]:
```suggestion
    - description: Fix handling of differently formatted data.
```

**Review by efd6 — COMMENTED** [review]
Is there a test case that I'm missing that exercises the `json.state instanceof String` case?

**Comment by chrisberkhout** [comment]
> Is there a test case that I'm missing that exercises the `json.state instanceof String` case?

@efd6 Yes, that's the existing case. There are 5 examples (`"state":"`) in [`data_stream/threat_response/_dev/test/pipeline/test-threat-response.log`](https://github.com/elastic/integrations/blob/main/packages/tanium/data_stream/threat_response/_dev/test/pipeline/test-threat-response.log).

**Review by efd6 — APPROVED** [review]
Thanks

**Pull request merged**